chain/network/src/peer_manager.rs

```diff
@@ -673,9 +674,13 @@ impl PeerManagerActor {
         }
 
         if self.is_outbound_bootstrap_needed() {
-            let mut ignore_list = self.outgoing_peers.clone();
-            ignore_list.insert(self.peer_id.clone());
-            if let Some(peer_info) = self.sample_random_peer(&ignore_list) {
+            if let Some(peer_info) = self.sample_random_peer(|peer_state| {
+                // Ignore connecting to ourself
+                self.peer_id == peer_state.peer_info.id
+                    || self.config.addr == peer_state.peer_info.addr
```

Comment:
I don't understand why `self.peer_id` wasn't enough actually? is it possible that address is the same but id is different?...

Reply:
It is possible when we change peer_id. We learn about our previous version that is still stored by other nodes in the network. This can happen on network updates if we nuke `~/.near` folder